src/types/pre_checkout_query.rs

```diff
@@ -1,12 +1,36 @@
 use crate::types::{OrderInfo, User};
 
-#[derive(Debug, Deserialize, Hash, PartialEq, Eq, Clone)]
+/// This object contains information about an incoming pre-checkout query.
+///
+/// [The official docs](https://core.telegram.org/bots/api#precheckoutquery).
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
 pub struct PreCheckoutQuery {
+    /// Unique query identifier.
     pub id: String,
+
+    /// User who sent the query.
     pub from: User,
+
+    /// Three-letter ISO 4217 [currency] code.
+    ///
+    /// [currency]: https://core.telegram.org/bots/payments#supported-currencies
     pub currency: String,
```

Comment:
Should we add `// TODO: checked ISO 4217 type` there?

Reply:
And an Issue.